cmd/wallet-js-sdk/src/oidc/presentation/openid4vp.js

```diff
@@ -120,6 +129,182 @@ export class OpenID4VP {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: presentation_submission is missing"
+      );
+    } else if (!presentationQuery.type) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: type is missing"
+      );
+    } else if (!presentationQuery.verifiableCredential) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: verifiableCredential is missing"
+      );
+    } else if (!issuer) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: issuer is missing"
+      );
+    } else if (!expiry) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: expiry is missing"
+      );
+    }
+
+    const header = new Object();
+    Object.defineProperty(header, "alg", {
+      value: alg,
+    });
+    Object.defineProperty(header, "kid", {
+      value: issuer,
+    });
+    Object.defineProperty(header, "typ", {
+      value: "JWT",
+    });
+
+    const idToken = await generateIdToken(
+      presentationQuery.presentation_submission,
+      header,
+      expiry
+    );
+
+    const vp = new Object();
+    Object.defineProperty(payload, "@context", {
+      value: presentationQuery["@context"],
+    });
+    Object.defineProperty(payload, "type", {
+      value: presentationQuery.type,
+    });
+    Object.defineProperty(payload, "verifiableCredential", {
+      value: presentationQuery.verifiableCredential,
+    });
+
+    const vpToken = await generateVpToken(header, vp, expiry);
+
+    const authRequest = new URLSearchParams();
+    authRequest.append("id_token", idToken);
+    authRequest.append("vp_token", vpToken);
+
+    return await axios.post(this.redirect_uri, authRequest).catch((e) => {
+      throw new Error("Error submitting OIDC presentation:", e);
+    });
+  }
+}
+
+/**
+ * generateIdToken generates an ID Token for the presentation submission request
+ * @param {string} kid - consumer's verification method's kid.
+ * @param {Object} presentationSubmission - presentation submission.
+ * @param {Object} header - header for the ID Token.
+ * @param {number} expiry - time in seconds representing the expiry of the token.
+ *
+ *
+ * @returns {Promise<string>} - a promise resolving with a string containing ID Token or an error.
+ */
+async function generateIdToken(kid, presentationSubmission, header, expiry) {
+  if (!kid) {
+    throw new TypeError("Error generating ID Token: kid cannot be empty");
+  } else if (!presentationSubmission) {
+    throw new TypeError(
+      "Error generating ID Token: presentationSubmission cannot be empty"
+    );
+  } else if (!header) {
+    throw new TypeError("Error generating ID Token: header cannot be empty");
+  } else if (!expiry) {
+    throw new TypeError("Error generating ID Token: expiry cannot be empty");
+  }
+
+  const vpToken = new Object();
+  Object.defineProperty(vpToken, "presentation_submission", {
```

Comment:
just curious about this syntax,

Why can't

``
var vpToken = {}
vpToken. presentation_submission = presentationSubmission

``

Reply:
This way, the properties inside the `vpToken` object are not modifiable

Reply:
I cannot use this syntax to create property `@context` on the object, which is required for https://github.com/trustbloc/agent-sdk/pull/453/files#diff-73c781567add49e4fe09e28cea2e4f4530d4999259ce327f63a89dc043d3d68dR199

I figured it'd be better to be consistent with the approach I take

Reply:
vpToken["@context"] = ctx